### ISSUE TITLE
Ubuntu 14.04 make command issue

### ISSUE THREAD

**Issue by kcraptor82** [xenserver/buildroot#688]
/buildroot# make
Makefile:7: /usr/share/planex/Makefile.rules: No such file or directory
make: **\* No rule to make target ` /usr/share/planex/Makefile.rules' . Stop.
/buildroot#

Have installed planex-0.7.2, what is wrong?

**Comment by johnkeates** [comment]
It's broken because nobody seems to have updated this project for Debian-based distro's for a while. To get at least planex working: 
- get https://github.com/xenserver/planex/archive/v0.7.3.tar.gz 
- change the name to planex_0.7.2.orig.tar.gz 
- expand it but keep the tgz file
- cd into the planex directory, and execute: "dpkg-buildpackage -d"
- install the resulting python-planex_0.7.2-1_all.deb using dpkg

There is another issue, a SPEC file for one of the ocaml modules has some hardcoded RedHat path in it that doesn't exist in Debian so that errors out, and when building libre-ocaml it can't seem to find 'threads' even when it's actually there.

I'm trying to get to a working setup in Debian Jessie, send patches and then have it working on other Debian-alike systems too. Currently, I'm dumping my work at https://github.com/johnkeates/xenbuilder-vagrant which uses Vagrant to setup a virtual Jessie box, and use a few SaltStack configuration management files to get the files and dependencies in place. It's a work in progress but maybe with some more people we can get this building on a sane Debian-alike OS again.

**Comment by tlehtil** [comment]
Tried to build a package on ubuntu 16.04 but got following error:
pyversions: missing X(S)-Python-Version in control file, fall back to debian/pyversions
pyversions: missing debian/pyversions file, fall back to supported versions
        python setup.py clean -a
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    from setuptools import setup
ImportError: No module named setuptools
dh_auto_clean: python setup.py clean -a returned exit code 1
debian/rules:7: recipe for target 'clean' failed
make: *** [clean] Error 1
dpkg-buildpackage: error: debian/rules clean gave error exit status 2

EDIT. nvm, fixed the problem by installing following packages first: 
python-rpm python-setuptools python-argparse rpm
but got new error:
/tmp/buildroot# make
....
File not found: _build/SRPMS/ocaml-findlib_1.5.5-1.dsc
/usr/share/planex/Makefile.rules:141: recipe for target '_build/RPMS/ocaml-findlib_1.5.5-1_amd64.deb' failed
make: *** [_build/RPMS/ocaml-findlib_1.5.5-1_amd64.deb] Error 1

**Comment by johnkeates** [comment]
Yeah, the maintainers/owners do not support Debian & Ubuntu, and planex doesn't either. IIRC, they even removed what bits were there to actually build anything at this point.

I do think it can be worked around since it's mostly OCaml stuff and planex/rpm doesn't have anything to do with it. What those two basically do is resolve dependencies between the packages and them build them in order as RPMs. If the spec files can be cloned and made into control files (and preinst/postinst files), and some sort of build order is made, you'd have a working setup. The problem is that you'd have to do that for every change, so that's not very efficient. A better solution would be working on planex and maintaining in-tree debian support, but that'd have to be agreed upon by Citrix/XenServer FOSS in order to happen, and maintainers would be needed as well.
